### PR TITLE
new toolchains + fixes 

### PR DESCRIPTION
**Pull request by headupinclouds**
* added 8.2 and 8.4 iOS SDK toolchains w/ libc++
* fix typo in ANDROID_TOOLCHAIN_NAME for arm-linux-androidabi-clang3.5 toolchain
* change dSYM to FMWK in dynamic framework info.plist template (see Core Foundation Keys CFBundlePackageType in https://developer.apple.com/library/ios/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html)

### CODE REVIEW

File: ios-8-2-libcxx.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+set(MACOSX_BUNDLE_GUI_IDENTIFIER com.example)
+set(CMAKE_MACOSX_BUNDLE YES)
+set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer")
+
+set(IPHONEOS_ARCHS armv7;armv7s;arm64)
+set(IPHONESIMULATOR_ARCHS i386;x86_64)
+
+include("${CMAKE_CURRENT_LIST_DIR}/compiler/xcode.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/os/iphone.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/library/std/libcxx.cmake")
```
Comment (ruslo) [inline ios-8-2-libcxx.cmake]: Do we really need this one? I think libc++ used by default in iOS Xcode projects?

File: scripts/Info.plist [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <key>CFBundleInfoDictionaryVersion</key>
     <string>6.0</string>
     <key>CFBundlePackageType</key>
-    <string>dSYM</string>
+    <string>FMWK</string>
```
Comment (ruslo) [inline scripts/Info.plist]: One more reason to implement this in CMake. Usually this file can be tuned by user's variables. Just thought aloud.

**Comment by headupinclouds** [comment]
I just realized this was never merged.  Shall I submit a separate pull request for a default of: `<string>FMWK</string>`.  Without this the dynamic frameworks would crash when called by another framework in an iOS project.

I'm also adding the create_framework.py mods to allow user specified plist files, but I think it would be a good idea to have this fix in place.

**Comment by headupinclouds** [comment]
resubmitted just the FMWK mod in pull request #41.  closing this one, which has conflicts.

**Pull request closed without merging**